scalafix-core/src/main/scala/scalafix/rewrite/NoAutoTupling.scala

```diff
@@ -15,32 +15,68 @@ case class NoAutoTupling(mirror: Mirror) extends SemanticRewrite(mirror) {
     ctx.addLeft(args.head.tokens.head, "(") +
       ctx.addRight(args.last.tokens.last, ")")
 
+  private[this] def fixMethods(ctx: RewriteCtx,
+                               jvmSignature: String,
+                               argss: Seq[Seq[Term.Arg]],
+                               fixed: mutable.Set[Seq[Term.Arg]]): Patch = {
+    val argListSignatures =
+      jvmSignature.stripPrefix("(").takeWhile(_ != ')').split(';')
+    argListSignatures
+      .zip(argss)
+      .foldLeft(Patch.empty) {
+        case (patches, (argListSignature, args)) =>
+          if (!fixed(args) && singleTuplePattern
+                .matcher(argListSignature)
+                .matches && args.length > 1) {
+            fixed += args // dirty hack, see explanation above
+            patches + addWrappingParens(ctx, args)
+          } else {
+            patches
+          }
+      }
+  }
+
+  private[this] def fixFunctions(ctx: RewriteCtx,
+                                 symbol: Symbol,
+                                 argss: Seq[Seq[Term.Arg]],
+                                 fixed: mutable.Set[Seq[Term.Arg]],
+                                 mirror: Mirror): Patch =
+    mirror.database.denotations
+      .get(symbol)
+      .map { denot =>
+        val argListSignatures =
+          denot.info.split("=>").map(_.trim).toList.dropRight(1)
+        argListSignatures
+          .zip(argss)
+          .foldLeft(Patch.empty) {
+            case (patches, (argListSignature, args)) =>
+              if (!fixed(args) && argListSignature.startsWith("((") && argListSignature
```

Comment:
Pro tip, scalafmt does not force line breaks on infix operators so you have to it yourself ;) http://scalameta.org/scalafmt/#Infixapplications